core/unit_test/cuda/TestCuda_ReducerViewSizeLimit.cpp

```diff
@@ -142,8 +148,90 @@ struct ReduceViewSizeLimitTester {
   }
 };
 
+struct TeamSizeTester {
+  void run_test_team_size(Kokkos::AUTO_t teamSizeRequested,
+                          Kokkos::AUTO_t vectorLengthRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested,
+                                              vectorLengthRequested);
+    const auto maxTeamSize = myTeam.team_size_max(FunctorFor<policy_type>(),
+                                                  Kokkos::ParallelForTag());
+
+    const auto recommendedTeamSize = myTeam.team_size_recommended(
+        FunctorFor<policy_type>(), Kokkos::ParallelForTag());
+    EXPECT_GT(maxTeamSize, 0);
+  }
+
+  void run_test_team_size(Kokkos::AUTO_t teamSizeRequested,
+                          int vectorLengthRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested,
+                                              vectorLengthRequested);
+    const auto maxTeamSize = myTeam.team_size_max(FunctorFor<policy_type>(),
+                                                  Kokkos::ParallelForTag());
+
+    const auto recommendedTeamSize = myTeam.team_size_recommended(
+        FunctorFor<policy_type>(), Kokkos::ParallelForTag());
+    EXPECT_GT(maxTeamSize, 0);
+  }
+
+  void run_test_team_size(int teamSizeRequested,
+                          Kokkos::AUTO_t vectorLengthRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested,
+                                              vectorLengthRequested);
+    const auto maxTeamSize = myTeam.team_size_max(FunctorFor<policy_type>(),
+                                                  Kokkos::ParallelForTag());
+    EXPECT_GT(maxTeamSize, 0);
+  };
+
+  void run_test_team_size(int teamSizeRequested, int vectorLengthRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested,
+                                              vectorLengthRequested);
+    const auto maxTeamSize = myTeam.team_size_max(FunctorFor<policy_type>(),
+                                                  Kokkos::ParallelForTag());
+    EXPECT_GT(maxTeamSize, 0);
+  }
+
+  void run_test_team_size(Kokkos::AUTO_t teamSizeRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested);
+    const auto maxTeamSize = myTeam.team_size_max(FunctorFor<policy_type>(),
+                                                  Kokkos::ParallelForTag());
+    EXPECT_GT(maxTeamSize, 0);
+  }
+
+  void run_test_team_size(int teamSizeRequested) {
+    Kokkos::TeamPolicy<TEST_EXECSPACE> myTeam(10, teamSizeRequested);
+    const auto maxTeamSize = myTeam.team_size_max(FunctorFor<policy_type>(),
+                                                  Kokkos::ParallelForTag());
+    EXPECT_GT(maxTeamSize, 0);
+  }
+};
+
 }  // namespace Impl
 
+TEST(cuda, testTeamSizeMaxisPositive) {
+  Impl::TeamSizeTester teamSizeTester;
+
+  teamSizeTester.run_test_team_size(Kokkos::AUTO, Kokkos::AUTO);
+  teamSizeTester.run_test_team_size(Kokkos::AUTO, -1);
```

Comment:
Passing `-1` is a precondition violation in my opinion.
I am not sure what this is testing.

Reply:
I think a test can verify a that a precondition is violated. The API of TeamPolicy allows to construct it with this value. No assert are done in the code to prevent this precondition violation.  In addition, the issue #3371  raised an error  "if a CUDA TeamPolicy has a negative vector length" which can be the case with this kind of TeamPolicy construction.
@fnrizzi what do you think ?

Reply:
I don't see a point in testing for user-set negative team sizes or vector lengths either but we should check that `AUTO` works.

Reply:
I agree that -1 is precond violation, so should test for AUTO only

Reply:
:+1: b19d5adbbea814d6b32a0507a68744601d710d2e